pkg/accounting/projectusage.go

```diff
@@ -15,20 +23,119 @@ const (
 	ExpansionFactor = 3
 )
 
-// ExceedsAlphaUsage returns true if the storage or bandwidth usage limits have been exceeded
-// for a project in the past month (30 days). The usage limit is 25GB multiplied by the redundancy
-// expansion factor, so that the uplinks have a raw limit of 25GB.
+var (
+	// ErrProjectUsage general error for project usage
+	ErrProjectUsage = errs.Class("project usage error")
+)
+
+// ProjectUsage defines project usage
+type ProjectUsage struct {
+	projectAccountingDB ProjectAccounting
+	liveAccounting      live.Service
+	maxAlphaUsage       memory.Size
+}
+
+// NewProjectUsage created new instance of project usage service
+func NewProjectUsage(projectAccountingDB ProjectAccounting, liveAccounting live.Service, maxAlphaUsage memory.Size) *ProjectUsage {
+	return &ProjectUsage{
+		projectAccountingDB: projectAccountingDB,
+		liveAccounting:      liveAccounting,
+		maxAlphaUsage:       maxAlphaUsage,
+	}
+}
+
+// ExceedsBandwidthUsage returns true if the bandwidth usage limits have been exceeded
+// for a project in the past month (30 days). The usage limit is (e.g 25GB) multiplied by the redundancy
+// expansion factor, so that the uplinks have a raw limit.
 // TODO(jg): remove this code once we no longer need usage limiting for alpha release
 // Ref: https://storjlabs.atlassian.net/browse/V3-1274
-func ExceedsAlphaUsage(bandwidthGetTotal, inlineTotal, remoteTotal int64, maxAlphaUsageGB memory.Size) (bool, string) {
-	maxUsage := maxAlphaUsageGB.Int64() * int64(ExpansionFactor)
+func (usage *ProjectUsage) ExceedsBandwidthUsage(ctx context.Context, projectID uuid.UUID, bucketID []byte) (_ bool, limit memory.Size, err error) {
+	// Check if this projectID has exceeded alpha usage limits for bandwidth or storage used in the past month
+	// TODO: remove this code once we no longer need usage limiting for alpha release
+	// Ref: https://storjlabs.atlassian.net/browse/V3-1274
+
+	var group errgroup.Group
+	var bandwidthGetTotal int64
+	limit = usage.maxAlphaUsage
+
+	// TODO maybe project limit cache expiration time?
```

Comment:
what does "project limit cache expiration time" mean?

Reply:
its just an idea if we will need less DB operations. We can have cache where entry is invalidated after period of time (e.g. 1h). ofc maybe different type will fit better

Reply:
ah ok that makes sense. could we change the comment to be more clear? Something like `TODO(mn): to reduce db load, consider using a cache to retrieve the `project.UsageLimit` values`

Reply:
Sure, good point

Reply:
done